testing/python/transform/test_tilelang_transform_loop_vectorize.py

```
@@ -0,0 +1,133 @@
+@pytest.mark.parametrize("access_kind", ["load", "store", "implicit-row-stride"])
+def test_large_int32_affine_index_does_not_abort_layout_inference(access_kind):
+    """Regression for issue #3027's host-side vectorization analysis ICE."""
+    extent = 4
+    stride = 600_000_000
+
+    if access_kind == "load":
+
+        @T.prim_func
+        def main(
+            A: T.Tensor((2_000_000_000,), T.float32),
+            O: T.Tensor((extent,), T.float32),
```

Ruff (0.16.1)

[error] Ambiguous variable name: `O`

(E741)